### PR TITLE
add custom panel width

### PR DESCRIPTION
**Pull request by badescuga**
#### Pull request checklist

- [x] Addresses an existing issue: #0000
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [x] Documentation update

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Panel/Panel.tsx [hunk 1]
```diff
@@ -158,12 +158,18 @@ export class Panel extends BaseComponent<IPanelProps, IPanelState> {
...
+                ['ms-Panel--custom ' + styles.rootIsCustom]: type === PanelType.custom,
                 ['ms-Panel--hasCloseButton ' + styles.rootHasCloseButton]: hasCloseButton
               })
             }
           >
             { overlay }
             <FocusTrapZone
+              ref={ (item) => {
+                if (type === PanelType.custom && item != null) {
+                  item.refs.root.style.width = this.props.customWidth;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: We shouldn't be modifying the DOM like this, or reaching into refs (privates) or assuming root will be there...

instead, couldn't you just pass this in as a style?

```
<FocusTrapZone style={
  (condition) && { width: this.props.customWidth }
}>
```

Comment (badescuga) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: sure, updated!

File: packages/office-ui-fabric-react/src/components/Panel/Panel.tsx [hunk 2]
```diff
@@ -118,6 +119,7 @@ export class Panel extends BaseComponent<IPanelProps, IPanelState> {
     let isRTL = getRTL();
     let isOnRightSide = isRTL ? isLeft : !isLeft;
     const headerTextId = id + '-headerText';
+    const customWidthStyles = (type === PanelType.custom) ? { width: customWidth } : {};
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: Does this work on smaller screens? E.g. If the width is 888px, what happens on a 320px wide device?

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: Without media queries, it seems custom widths would be troublesome. We might be able to get away with setting that width, and then a max-width of 80% or something.

Comment (badescuga) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: sorry for the late response, was on vacation. 

IMO, custom width behaviour should be manually handled by the dev. There could be edge-cases where we want to cover more than 80% of the screen (e.g. cover the whole screen expect a left nav menu). Having a custom width that isn't actually custom seems a bit confusing.

File: common/changes/gabade-add-custom-panel-width_2017-03-27-15-49.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      "type": "minor"
+    },
+    {
+      "comment": "",
+      "packageName": "@uifabric/utilities",
+      "type": "none"
+    },
+    {
+      "packageName": "@uifabric/example-app-base",
+      "comment": "Panel: adding `customWidth` property and PanelType.custom value to support custom panel widths.",
```
Comment (dzearing) [inline common/changes/gabade-add-custom-panel-width_2017-03-27-15-49.json]: don't think you're modifying example-app-base here.

**Comment by mikewheaton** [comment]
Discussing offline if it's possible to achieve his requirements with CSS overrides. I'm concerned about adding a new prop here for custom widths if we're going to deprecate it soon in favor of customization as discussed in #332.

**Comment by badescuga** [comment]
If there's a design decision (deprecate Panel/deprecate some props/implement new custom panel), I can work on implementing it.

File: packages/office-ui-fabric-react/src/components/Panel/Panel.tsx (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Panel/Panel.tsx]: makes sense to me.

**Comment by dzearing** [comment]
I think i saw a mail thread earlier on this. i'll confirm this is fine tomorrow and then we'll merge.

**Comment by mikewheaton** [comment]
I think we should close this PR, and open a new one that implements customization that will work across breakpoints as discussed in #332. Simple width overrides like this are best handled with a custom CSS class.

**Comment by badescuga** [comment]
As mentioned offline, I can set it to work passing a custom class by using !important for width, but that's not a good pattern. Don't want to commit to changing a lot of css in the panel. The best solutions I see: 
1) use this PR fix
2) pass in React.CSSProperties through which we can set the width
3) Have a clear plan for #332 . Right now it seems to be wishful thinking (as it was opened back in September 2016). If there's a clear design decision on how we should tackle this (what's to be deprecated, etc), I can work on it.

**Comment by mikewheaton** [comment]
As discussed via Teams, we're going to go ahead with this change to make sure no one is blocked. #332 looks like the best long-term solution, but without a clear plan there I don't want to hold this up.

**Pull request merged**